RULES.md

```diff
@@ -74,7 +76,7 @@ def build_input_queue(
 ###### Model initialization
```

Comment:
The line just above this is describing the `build_input_queue` function, and it mentions

> ... and for PyTorch this should return tensors of shape `(global_batch_size, ...)`.

Is this still correct, or should it instead be `(per_device_batch_size, ...)`, which is the usual way that data loading is handled with multi-gpu training in PyTorch (with a DistributedSampler, so that each device gets different samples).

cc @runame

Reply:
It actually depends on whether DP or DDP is used. In the former case, the current statement `(global_batch_size, ...)` is correct; in the latter, `(per_device_batch_size, ...)` is correct.

Reply:
We should really only be using DDP

Reply:
As discussed in the meeting, I will create a second PR with all remaining rules changes where we can address this.
I can change it to `per_device_batch_size`. Should I also add something about DDP?

Reply:
Maybe just add something like "(assuming DDP is used)". I agree that in practice only DDP should be used, but it is useful for debugging to also support DP in the code.

Reply:
Done.